### ISSUE TITLE
check output not showing up in `problems` when `rust-analyzer.rust.analyzerTargetDir` is set

### ISSUE THREAD

**Issue by BLucky-gh** [rust-lang/rust-analyzer#16353]
rust-analyzer version: 0.3.1799-standalone (af4010184 2024-01-07)

Krustc version: rustc 1.75.0 (82e1608df 2023-12-21)

relevant settings: `"rust-analyzer.rust.analyzerTargetDir": "target/rust-analyzer",`

When the target dir is set, none of the issues show up, for example if I set my `main.rs` to be
```rust
fn main(){
    let foo = 3;
}
```

`cargo check` will warn about the unused var, and if I unset `analyzerTargetDir` it shows up in `Problems` in vscode, but if I set it, it doesn't show up

**Comment by JeanMertz** [comment]
@BLucky-gh did you ever find out what was causing this? I'm running into the exact same situation in Neovim.

I can disable `analyzerTargetDir` (which I've set to `true`) to get the diagnostics to appear again, but that obviously brings the downside of "compilation fighting" between RA and other Rust tools.

**Comment by BLucky-gh** [comment]
Nope, I completely forgot about this bug since I had to reinstall vscode and lost my config. I haven't had to deal with projects as large as the one that prompted me to set `analyzerTargetDir` since, so RA and cargo fighting hasn't been noticeable and I just forgot

**Comment by JeanMertz** [comment]
For the record, I seem to have resolved the issue (for now) by nuking the target directory, the running `cargo check` first, before starting my editor with RA.
